packages/foundation/src/atoms/core/colors/brand.css

```diff
@@ -1,35 +1,38 @@
-:root {
+[class^="o-ui-"] {
```

Comment:
I don't think something in foundation should be scope with "o-ui". I do see our foundation as our current implementation or our future design tokens so it should be developed under the same principles.

Reply:
We can't leave this at root for 2 reasons, first root could clash with other imported CSS using  custom properties at root too, second using root has a drawback ` we can’t rely on inheritance for functions that need to be re-calculated. That’s because each property value is computed once per element (in our case, the html root element), and then the computed value is inherited. By defining our function at the document root, we don’t make the entire function available to descendants — only the computed result of our function.`

tl:dr: for a custom property to be available everywhere and used in other custom properties declarations it needs to be attached to a common class and not root.

Reply:
I don't think the first argument is a valid one. Since this is a foundation, this is expected to be available. It is the responsability of the consumer apps to ensure their CSS properties doesn't clash with the foundation.

Do you have a specific use case which we currently have that's doesn't work without scoping these properties?

Reply:
If we must scope these properties, I believe the way to go is to prefix the property names with "o-ui", not scoping to [class^="o-ui-"].

Reply:
`
.o-ui-checkbox-invalid .o-ui-checkbox-box::before {
    --o-ui-focus-ring-color: var(--critical-border);
}
`

here is an example, redefining the focus ring color that is used here `--o-ui-focus-ring-inset-md: 0 0 0 2px var(--o-ui-focus-ring-color);` that way you don't have to remember / redeclare the ` 0 0 0 2px` part and by mistake change the default value.